maxpool.h

```diff
@@ -468,11 +469,10 @@ template<
     unsigned int NumTop,
         typename In_T,
     typename Out_T>
-void LabelSelect_Batch(stream<ap_uint<PECount * In_T::width> > & in,
-        stream<Out_T> & out, const unsigned int numReps) { 
+void LabelSelect_Batch(hls::stream<ap_uint<PECount * In_T::width> > & in,
+        hls::stream<Out_T> & out, const unsigned int numReps) {
 
-  const Out_T Out_T_MAX_VAL = (Out_T(-1)<0)? ~(1<<(Out_T::width-1)) : ~(0);
-  CASSERT_DATAFLOW(Out_T_MAX_VAL >= NumClasses-1);
+  static_assert(clog2<NumClasses>::value <= Out_T::width - Out_T::sign_flag);
```

Comment:
@maltanar: This change appears to maintain the desired semantics of the previous assertion. However, I am not clear why that original assertion actually only considers the non-negative coding space of the output. Shouldn't it be irrelevant whether this MSB a sign bit or not?